pyccel/parser/semantic.py

```diff
@@ -870,6 +904,92 @@ def _handle_function(self, expr, func, args, **settings):
                             expr, func.is_elemental)
             return new_expr
 
+    def _handle_kernel(self, expr, func, args, **settings):
+        """
+        Create a FunctionCall or an instance of a PyccelInternalFunction
+        from the function information and arguments
+
+        Parameters
+        ==========
+        expr : PyccelAstNode
+               The expression where this call is found (used for error output)
+        func : FunctionDef instance, Interface instance or PyccelInternalFunction type
+               The function being called
+        args : tuple
+               The arguments passed to the function
+
+        Returns
+        =======
+        new_expr : FunctionCall or PyccelInternalFunction
+        """
+        if 'kernel' not in func.decorators:
+            errors.report(INVALID_FUNCTION_CALL,
+                        symbol = expr,
+                        severity = 'fatal')
+        if isinstance(func, PyccelFunctionDef):
+            func = func.cls_name
+            args, kwargs = split_positional_keyword_arguments(*args)
+            for a in args:
+                if getattr(a,'dtype',None) == 'tuple':
+                    self._infere_type(a, **settings)
+            for a in kwargs.values():
+                if getattr(a,'dtype',None) == 'tuple':
+                    self._infere_type(a, **settings)
```

Comment:
Using old function name:

```suggestion
                    self._infer_type(a, **settings)
            for a in kwargs.values():
                if getattr(a,'dtype',None) == 'tuple':
                    self._infer_type(a, **settings)
```

Reply:
This code is untested

Reply:
This code isn't reported as untested but I'm not sure how that can be true if it's using a function that doesn't exist. Something must be behaving weirdly